crates/swc/src/plugin.rs

```diff
@@ -41,24 +41,29 @@ pub struct PluginContext {
     pub env_name: String,
 }
 
+#[cfg(all(feature = "plugin", not(target_arch = "wasm32")))]
 pub fn plugins(
     resolver: CachingResolver<NodeModulesResolver>,
     config: crate::config::JscExperimental,
     plugin_context: PluginContext,
 ) -> impl Fold {
-    #[cfg(feature = "plugin")]
     {
         RustPlugins {
             resolver,
             plugins: config.plugins,
             plugin_context,
         }
     }
+}
 
-    #[cfg(not(feature = "plugin"))]
-    {
-        noop()
-    }
+#[cfg(all(feature = "plugin", target_arch = "wasm32"))]
+pub fn plugins(config: crate::config::JscExperimental, plugin_context: PluginContext) -> impl Fold {
```

Comment:
It seems that plugins API prepared only for JS, perhaps we should reconsider some things so that in the future it will be more extensible and be able to use CSS/HTML/Markdown/etc, just notes

Reply:
You're correct, current plugin is completely tailored for *transform* only, even among the js execution paths. For example, there's no *parse* plugin interfaces.

I'm trying to move each step forward, current state is making every runtime able to load plugin & execute at least. We started from transform, so completing it resolves major exceution paths (load, execute wasm plugin). From there we need to revisit / expand interfaces to conform with remaining possible paths we want to support transform.

Does it sounds ok to you?

Reply:
Sounds good :+1: